src/main/java/uk/gov/nationalarchives/pdi/step/jena/Util.java

```diff
@@ -58,10 +75,72 @@ public static String emptyIfNull(@Nullable final String s) {
         }
     }
 
+    /**
+     * Returns true if the String is either null or empty.
+     *
+     * @param s a String
+     *
+     * @return true if the String is null or empty, false otherwise.
+     */
     public static boolean isNullOrEmpty(@Nullable final String s) {
         return s == null || s.isEmpty();
     }
 
+    /**
+     * Returns true if the String is not-empty (or null).
+     *
+     * @param s a String
+     *
+     * @return true if the String is not null and not empty, false otherwise.
+     */
+    public static boolean isNotEmpty(@Nullable final String s) {
+        return s != null && !s.isEmpty();
+    }
+
+    /**
+     * Returns true if the List is either null or empty.
+     *
+     * @param l a List
+     *
+     * @return true if the List is null or empty, false otherwise.
+     */
+    public static <T> boolean isNullOrEmpty(@Nullable final List<T> l) {
+        return l == null || l.isEmpty();
+    }
+
+    /**
+     * Returns true if the Map is not-empty (or null).
+     *
+     * @param m a Map
+     *
+     * @return true if the Map is not null and not empty, false otherwise.
+     */
+    public static <K,V> boolean isNotEmpty(@Nullable final Map<K, V> m) {
+        return m != null && !m.isEmpty();
+    }
+
+    /**
+     * Returns true if the List is not-empty (or null).
+     *
+     * @param l a List
+     *
+     * @return true if the List is not null and not empty, false otherwise.
+     */
+    public static <T> boolean isNotEmpty(@Nullable final List<T> l) {
+        return l != null && !l.isEmpty();
+    }
+
+    /**
+     * Returns true if the Array is not-empty (or null).
+     *
+     * @param a an Array
+     *
+     * @return true if the Array is not null and not empty, false otherwise.
+     */
+    public static <T> boolean isNotEmpty(@Nullable final T[] a) {
+        return a != null && a.length > 0;
+    }
+
```

Comment:
Feels like there should be tests for all these methods to show they do what they say.

Reply:
I have now added the missing tests.